### PR TITLE
libScePad rework

### PR DESCRIPTION
**Pull request by igor725**

### CODE REVIEW

File: modules/libScePad/cconfig.cpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  std::string temp;
+  type.get_to(temp);
+  LOG_CRIT(L"Unknown controller type: %S", temp.c_str());
+  return ControllerType::Unknown;
+}
+
+uint32_t MapControllerKey(const std::string& key) {
+  LOG_USE_MODULE(libScePad_config);
+
+  if (key == "l3") {
```
Comment (SysRay) [inline modules/libScePad/cconfig.cpp:29]: static std::unordered::map<std::string_view, Controllerkey> ?

File: modules/libScePad/interfaces/ixip.cpp [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  if (m_state == ControllerState::Disconnected || m_state == ControllerState::Closed) return;
+  m_state = ControllerState::Closed;
+}
+
+bool XIPController::reconnect() {
+  return false;
+}
+
+uint32_t XIPController::getButtons() {
+  std::bitset<32> bits;
```
Comment (SysRay) [inline modules/libScePad/interfaces/ixip.cpp:60]: do you wan't to change the values later? bits is default = 0

File: modules/libScePad/types.h [hunk 3]
```diff
@@ -95,6 +95,49 @@ enum class SceFlightCapability {
   RADAR_CURSOR_STICK = 0x10,
 };
 
+enum ControllerKey : uint32_t {
```
Comment (SysRay) [inline modules/libScePad/types.h:98]: enum class

File: modules/libScePad/interfaces/ixip.cpp (hunk 2 above)
Comment (igor725) [reply modules/libScePad/interfaces/ixip.cpp:60]: Yep, just a stub to change these values later, but... Actually iirc XInput stores pressed buttons in `int` value too, so I'll remove this bc we just need to convert bitsets between XInput and libScePad

File: modules/libScePad/cconfig.cpp (hunk 1 above)
Comment (igor725) [reply modules/libScePad/cconfig.cpp:29]: I'm always forgetting about maps...

**Pull request merged**